// @ant-design/web3/dist/lib/connect-modal/components/GetWalletPanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _context = require("../context");
var _MainPanelHeader = _interopRequireDefault(require("./MainPanelHeader"));
var _antd = require("antd");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const GetWalletPanel = props => {
  const {
    walletList = []
  } = props;
  const [, contextHolder] = _antd.message.useMessage();
  const {
    prefixCls,
    updateSelectedWallet,
    updatePanelRoute,
    localeMessage
  } = (0, _react.useContext)(_context.connectModalContext);
  const list = /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [contextHolder, /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: `${prefixCls}-list`,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.List, {
        itemLayout: "horizontal",
        dataSource: walletList,
        renderItem: item => /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.List.Item, {
          className: `${prefixCls}-item`,
          actions: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Button, {
            type: "default",
            shape: "round",
            className: `${prefixCls}-get-wallet-btn`,
            onClick: () => {
              updateSelectedWallet(item);
              updatePanelRoute('wallet');
            },
            children: localeMessage.getWalletBtnText
          }, "get")],
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.List.Item.Meta, {
            avatar: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Avatar, {
              size: 48,
              shape: "square",
              src: item.icon,
              children: item.name[0]
            }),
            title: item.name,
            description: item.remark
          })
        })
      })
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: `${prefixCls}-info`,
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("h3", {
        children: localeMessage.getWalletPanelInfoTitle
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        children: localeMessage.getWalletPanelInfoDesc
      })]
    })]
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    className: `${prefixCls}-get-wallet-panel`,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_MainPanelHeader.default, {
      title: localeMessage.getWalletPanelTitle
    }), list]
  });
};
var _default = exports.default = GetWalletPanel;
