// @ant-design/web3/dist/lib/connect-button/connect-button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConnectButton = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _classnames = _interopRequireDefault(require("classnames"));
var _address = require("../address");
var _tooltip = require("./tooltip");
var _chainSelect = require("./chain-select");
var _profileModal = require("./profile-modal");
var _style = require("./style");
var _utils = require("../utils");
var _cryptoPrice = require("../crypto-price");
var _icons = require("@ant-design/icons");
var _useIntl = _interopRequireDefault(require("../hooks/useIntl"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const ConnectButton = props => {
  const {
    onConnectClick,
    onDisconnectClick,
    availableChains,
    onSwitchChain,
    tooltip,
    chain,
    account,
    avatar,
    profileModal = true,
    onMenuItemClick,
    actionsMenu = false,
    loading,
    onClick,
    balance,
    className,
    locale,
    ...restProps
  } = props;
  const intl = (0, _useIntl.default)('ConnectButton', locale);
  const {
    getPrefixCls
  } = (0, _react.useContext)(_antd.ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('web3-connect-button');
  const [profileOpen, setProfileOpen] = (0, _react.useState)(false);
  const {
    wrapSSR,
    hashId
  } = (0, _style.useStyle)(prefixCls);
  const [messageApi, contextHolder] = _antd.message.useMessage();
  const [showMenu, setShowMenu] = (0, _react.useState)(false);
  let buttonText = intl.getMessage(intl.messages.connect);
  if (account) {
    buttonText = account?.name && !balance ? account?.name : /*#__PURE__*/(0, _jsxRuntime.jsx)(_address.Address, {
      tooltip: false,
      ellipsis: true,
      address: account.address,
      children: balance ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_cryptoPrice.CryptoPrice, {
        icon: true,
        ...balance
      }) : undefined
    });
  }
  const buttonProps = {
    style: props.style,
    size: props.size,
    type: props.type,
    ghost: props.ghost,
    loading,
    className: (0, _classnames.default)(className, prefixCls, hashId),
    onClick: e => {
      setShowMenu(false);
      if (account && !profileOpen && profileModal) {
        setProfileOpen(true);
      } else {
        onConnectClick?.();
      }
      onClick?.(e);
    },
    ...restProps
  };
  const renderChainSelect = () => {
    if (availableChains && availableChains.length > 1) {
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(_chainSelect.ChainSelect, {
        hashId: hashId,
        onSwitchChain: onSwitchChain,
        currentChain: chain,
        chains: availableChains
      });
    }
    return null;
  };
  const chainSelect = renderChainSelect();
  const buttonInnerText = /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: `${prefixCls}-content`,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: `${prefixCls}-content-inner`,
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-text`,
        children: buttonText
      }), avatar && /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Divider, {
          type: "vertical"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: `${prefixCls}-avatar`,
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Avatar, {
            ...avatar
          })
        })]
      })]
    })
  });
  const buttonContent = chainSelect ? /*#__PURE__*/(0, _jsxRuntime.jsxs)(_antd.Space.Compact, {
    children: [chainSelect, /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Button, {
      ...buttonProps,
      children: buttonInnerText
    })]
  }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Button, {
    ...buttonProps,
    children: buttonInnerText
  });
  const profileModalContent = /*#__PURE__*/(0, _jsxRuntime.jsx)(_profileModal.ProfileModal, {
    intl: intl,
    open: profileOpen,
    __hashId__: hashId,
    onDisconnect: () => {
      setProfileOpen(false);
      onDisconnectClick?.();
    },
    onClose: () => {
      setProfileOpen(false);
    },
    address: account?.address,
    name: account?.name,
    avatar: avatar ?? {
      icon: chain?.icon ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-chain-icon`,
        children: chain?.icon
      }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.UserOutlined, {
        className: `${prefixCls}-default-icon`
      })
    },
    balance: balance,
    modalProps: typeof profileModal === 'object' ? profileModal : undefined
  });
  let content = buttonContent;
  const defaultMenuItems = (0, _react.useMemo)(() => [{
    label: 'Copy Address',
    key: 'copyAddress',
    onClick: () => {
      setProfileOpen(false);
      if (account?.address) {
        (0, _utils.writeCopyText)(account?.address).then(() => {
          messageApi.success('Address Copied!');
        });
      }
    },
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.CopyOutlined, {})
  }, {
    label: intl.getMessage(intl.messages.disconnect),
    key: 'disconnect',
    onClick: () => {
      setProfileOpen(false);
      onDisconnectClick?.();
    },
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.LoginOutlined, {})
  }], [account?.address, messageApi, onDisconnectClick]);
  const mergedMenuItems = (0, _react.useMemo)(() => {
    if (!actionsMenu) {
      return [];
    }
    if (typeof actionsMenu === 'boolean') {
      return account ? defaultMenuItems : [];
    }
    if (actionsMenu.items) {
      return actionsMenu.items;
    }
    const combinedItems = account ? actionsMenu.extraItems ? [...actionsMenu.extraItems, ...(account ? defaultMenuItems : [])] : account ? defaultMenuItems : [] : actionsMenu.extraItems || [];
    return combinedItems;
  }, [actionsMenu, defaultMenuItems, account]);
  if (mergedMenuItems.length > 0) {
    content = /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Dropdown, {
      open: showMenu,
      onOpenChange: setShowMenu,
      menu: {
        items: mergedMenuItems,
        onClick: onMenuItemClick
      },
      children: buttonContent
    });
  }
  const mergedTooltipCopyable = typeof tooltip === 'object' ? tooltip.copyable !== false : !!tooltip;
  let tooltipTitle = tooltip && account?.address ? (0, _utils.fillWith0x)(account?.address) : '';
  if (typeof tooltip === 'object' && typeof tooltip.title === 'string') {
    tooltipTitle = tooltip.title;
  }
  const main = /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [contextHolder, tooltipTitle ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_tooltip.ConnectButtonTooltip, {
      intl: intl,
      copyable: mergedTooltipCopyable,
      title: tooltipTitle,
      prefixCls: prefixCls,
      __hashId__: hashId,
      ...(typeof tooltip === 'object' ? tooltip : {}),
      children: content
    }) : content, profileModalContent]
  });
  return wrapSSR(main);
};
exports.ConnectButton = ConnectButton;
ConnectButton.displayName = 'ConnectButton';
