// @ant-design/web3/dist/lib/crypto-price/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CryptoPrice = void 0;
var _balance = require("./balance");
var _antd = require("antd");
var _react = require("react");
var _style = require("./style");
var _web3Assets = require("@ant-design/web3-assets");
var _useProvider = _interopRequireDefault(require("../hooks/useProvider"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const CryptoPrice = props => {
  const {
    getPrefixCls
  } = (0, _react.useContext)(_antd.ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('web3-crypto-balance');
  const {
    wrapSSR,
    hashId
  } = (0, _style.useStyle)(prefixCls);
  const {
    className,
    chain: userConfigChain
  } = props;
  const {
    chain = _web3Assets.Mainnet
  } = (0, _useProvider.default)({
    chain: userConfigChain
  });
  const {
    value = 0n,
    symbol = chain.nativeCurrency?.symbol ?? 'ETH',
    decimals = chain.nativeCurrency?.decimals ?? 18,
    fixed,
    icon = false
  } = props;
  return wrapSSR( /*#__PURE__*/(0, _jsxRuntime.jsx)(_balance.CryptoPriceBalance, {
    fixed: fixed,
    value: value,
    symbol: symbol,
    decimals: decimals,
    className: className,
    hashId: hashId,
    icon: icon === true ? chain.nativeCurrency?.icon : icon
  }));
};
exports.CryptoPrice = CryptoPrice;
CryptoPrice.displayName = 'CryptoPrice';
