// @ant-design/web3/dist/esm/connect-button/chain-select.js
import React, { useContext } from 'react';
import { Dropdown, Space, ConfigProvider, Button } from 'antd';
import classNames from 'classnames';
import { DownOutlined } from '@ant-design/icons';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var ChainSelect = function ChainSelect(_ref) {
  var className = _ref.className,
    onSwitchChain = _ref.onSwitchChain,
    style = _ref.style,
    chains = _ref.chains,
    hashId = _ref.hashId,
    currentChain = _ref.currentChain;
  var _useContext = useContext(ConfigProvider.ConfigContext),
    getPrefixCls = _useContext.getPrefixCls;
  var prefixCls = getPrefixCls('web3-connect-button-chain-select');
  return /*#__PURE__*/_jsx(Button, {
    className: "".concat(prefixCls, "-button"),
    children: /*#__PURE__*/_jsx(Dropdown, {
      className: classNames(className, hashId, prefixCls),
      overlayClassName: "".concat(prefixCls, "-dropdown"),
      menu: {
        items: chains.map(function (chain) {
          return {
            key: chain.id,
            label: chain.name,
            icon: chain.icon,
            onClick: function onClick() {
              onSwitchChain === null || onSwitchChain === void 0 || onSwitchChain(chain);
            }
          };
        })
      },
      trigger: ['click'],
      children: /*#__PURE__*/_jsx("div", {
        className: "".concat(prefixCls, "-placeholder"),
        style: style,
        children: /*#__PURE__*/_jsxs(Space, {
          children: [currentChain === null || currentChain === void 0 ? void 0 : currentChain.icon, currentChain === null || currentChain === void 0 ? void 0 : currentChain.name, /*#__PURE__*/_jsx(DownOutlined, {})]
        })
      })
    })
  });
};
ChainSelect.displayName = 'ChainSelect';
