// @ant-design/web3/dist/esm/utils/test-utils.js
var oriClipboard = window.navigator.clipboard;
// 代理 navigator.clipboard 方法用于测试复制文本到粘贴板的功能
export var mockClipboard = function mockClipboard() {
  var clipboard = {
    text: '',
    writeText: function writeText(text) {
      clipboard.text = text;
      return Promise.resolve();
    },
    readText: function readText() {
      return Promise.resolve(clipboard.text);
    }
  };
  Object.defineProperty(window, 'navigator', {
    value: {
      clipboard: clipboard
    }
  });
  return function () {
    Object.defineProperty(window, 'navigator', {
      value: {
        clipboard: oriClipboard
      }
    });
  };
};
export var browsers = {
  Chrome: 'chrome',
  Edge: 'edge',
  Firefox: 'firefox',
  Safari: 'safari'
};
export function mockBrowser(browser) {
  var oriUserAgent = window.navigator.userAgent;
  Object.defineProperty(window, 'navigator', {
    value: {
      userAgent: browsers[browser]
    }
  });
  return function () {
    Object.defineProperty(window, 'navigator', {
      value: {
        userAgent: oriUserAgent
      }
    });
  };
}
