// @ant-design/web3/dist/lib/nft-card/NFTCard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _style = require("./style");
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _antd = require("antd");
var _icons = _interopRequireDefault(require("@ant-design/icons"));
var _useNFT = _interopRequireDefault(require("../hooks/useNFT"));
var _useToken = _interopRequireDefault(require("antd/es/theme/useToken"));
var _tool = require("../utils/tool");
var _web3Common = require("@ant-design/web3-common");
var _cryptoPrice = require("../crypto-price");
var _useIntl = _interopRequireDefault(require("../hooks/useIntl"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var HeartSvg = function HeartSvg(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("svg", {
    ...props,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("path", {
      d: "M21.633 6.647a6.095 6.095 0 0 0-1.334-1.94A6.219 6.219 0 0 0 15.93 2.93 6.26 6.26 0 0 0 12 4.315 6.26 6.26 0 0 0 8.07 2.93 6.219 6.219 0 0 0 3.7 4.706a6.057 6.057 0 0 0-1.825 4.33c0 .78.16 1.593.476 2.42.265.692.644 1.409 1.13 2.133.768 1.146 1.825 2.341 3.138 3.553a35.08 35.08 0 0 0 4.42 3.453l.556.356a.753.753 0 0 0 .808 0l.556-.356a35.534 35.534 0 0 0 4.42-3.453c1.312-1.212 2.37-2.407 3.138-3.553a10.72 10.72 0 0 0 1.13-2.133c.316-.827.476-1.64.476-2.42a5.968 5.968 0 0 0-.49-2.39ZM12 19.097S3.656 13.75 3.656 9.035c0-2.388 1.976-4.324 4.414-4.324 1.713 0 3.199.956 3.93 2.353a4.426 4.426 0 0 1 3.93-2.353c2.438 0 4.414 1.936 4.414 4.324 0 4.716-8.344 10.062-8.344 10.062Z",
      fill: "#000",
      fillRule: "nonzero",
      fillOpacity: ".723",
      opacity: ".85"
    })
  });
};
HeartSvg.defaultProps = {
  width: "24",
  height: "24",
  xmlns: "http://www.w3.org/2000/svg"
};
var HeartFilledSvg = function HeartFilledSvg(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("svg", {
    ...props,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("defs", {
      children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("linearGradient", {
        x1: "50%",
        y1: "0%",
        x2: "50%",
        y2: "100%",
        id: "ant-web3-icon-nft-card-icon-heart-filled-1",
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("stop", {
          stopColor: "#FF8B8B",
          offset: "0%"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("stop", {
          stopColor: "#FF1717",
          offset: "100%"
        })]
      }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("filter", {
        x: "-10%",
        y: "-11.1%",
        width: "120%",
        height: "122.2%",
        filterUnits: "objectBoundingBox",
        id: "ant-web3-icon-nft-card-icon-heart-filled-3",
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("feGaussianBlur", {
          stdDeviation: "1.5",
          in: "SourceAlpha",
          result: "shadowBlurInner1"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("feOffset", {
          dx: "1",
          dy: "1",
          in: "shadowBlurInner1",
          result: "shadowOffsetInner1"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("feComposite", {
          in: "shadowOffsetInner1",
          in2: "SourceAlpha",
          operator: "arithmetic",
          k2: "-1",
          k3: "1",
          result: "shadowInnerInner1"
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("feColorMatrix", {
          values: "0 0 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 0 0.746667395 0",
          in: "shadowInnerInner1"
        })]
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("path", {
        d: "M36.516 21.689a6.05 6.05 0 0 0-1.317-1.926 6.12 6.12 0 0 0-1.945-1.29 6.118 6.118 0 0 0-2.37-.473A6.162 6.162 0 0 0 27 19.375 6.162 6.162 0 0 0 23.12 18c-.822 0-1.618.158-2.37.472a6.106 6.106 0 0 0-3.263 3.217 5.988 5.988 0 0 0-.486 2.37c0 .774.157 1.581.47 2.402.262.686.637 1.398 1.116 2.117.76 1.137 1.803 2.323 3.1 3.526a34.67 34.67 0 0 0 4.366 3.426l.549.353a.74.74 0 0 0 .798 0l.549-.353a35.117 35.117 0 0 0 4.366-3.426c1.297-1.203 2.34-2.389 3.1-3.526.48-.719.857-1.43 1.116-2.117.313-.82.47-1.628.47-2.402a5.946 5.946 0 0 0-.484-2.37Z",
        id: "a"
      })]
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("g", {
      transform: "translate(-17 -18)",
      fillRule: "nonzero",
      fill: "none",
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("use", {
        fill: "url(#ant-web3-icon-nft-card-icon-heart-filled-1)",
        xlinkHref: "#a"
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("use", {
        fill: "#000",
        filter: "url(#ant-web3-icon-nft-card-icon-heart-filled-3)",
        xlinkHref: "#a"
      })]
    })]
  });
};
HeartFilledSvg.defaultProps = {
  width: "20",
  height: "18",
  xmlns: "http://www.w3.org/2000/svg",
  xmlnsXlink: "http://www.w3.org/1999/xlink"
};
const customizePrefixCls = 'ant-nft-card';
const CardSkeleton = ({
  children,
  loading,
  prefixCls
}) => {
  if (loading) {
    return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_antd.Space, {
      direction: "vertical",
      className: (0, _classnames.default)(`${prefixCls}-wrap`),
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Skeleton.Image, {
        active: true,
        className: `${prefixCls}-content`
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Skeleton, {
        active: true,
        className: `${prefixCls}-body`
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Skeleton.Button, {
        block: true,
        active: true,
        className: `${prefixCls}-button`
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Divider, {
        dashed: true,
        style: {
          marginBlock: 0
        }
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Skeleton.Input, {
        active: true,
        className: `${prefixCls}-footer`
      })]
    });
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_jsxRuntime.Fragment, {
    children: children
  });
};
const NFTCard = ({
  style,
  antdImageProps,
  className,
  type = 'default',
  address,
  tokenId,
  price,
  like: likeConfig,
  showAction,
  actionText,
  footer,
  onActionClick,
  getNFTMetadata,
  locale,
  ...metadataProps
}) => {
  const {
    liked,
    totalLikes = 0,
    onLikeChange
  } = likeConfig || {};
  const [, token] = (0, _useToken.default)();
  const {
    metadata,
    loading
  } = (0, _useNFT.default)(address, (0, _web3Common.parseNumberToBigint)(tokenId), getNFTMetadata);
  const {
    name = metadata.name,
    image = metadata.image,
    description = metadata.description
  } = metadataProps;
  const {
    messages
  } = (0, _useIntl.default)('NFTCard', locale);
  const {
    getPrefixCls
  } = _react.default.useContext(_antd.ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('nft-card', customizePrefixCls);
  //================== Style ==================
  const {
    wrapSSR,
    hashId
  } = (0, _style.useStyle)(prefixCls);
  const mergeCls = (0, _classnames.default)(`${prefixCls}-container`, {
    [`${prefixCls}-pithy`]: type === 'pithy',
    [`${prefixCls}-theme-dark`]: (0, _tool.isDarkTheme)(token)
  }, className, hashId);
  const [like, setLike] = (0, _react.useState)(false);
  const handleLikeChange = () => {
    const likeValue = !like;
    if (liked === undefined) {
      setLike(likeValue);
    }
    onLikeChange?.(likeValue);
  };
  const iconLikeGroup = /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    className: `${prefixCls}-like-icon-group`,
    style: {
      position: 'absolute',
      top: 5,
      right: -2
    },
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.default, {
      component: HeartFilledSvg
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.default, {
      component: HeartFilledSvg,
      style: {
        transform: 'scale(0.4)',
        position: 'absolute',
        right: -8,
        top: -12
      }
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.default, {
      component: HeartFilledSvg,
      style: {
        transform: 'scale(0.3)',
        position: 'absolute',
        right: 10,
        top: -16
      }
    })]
  });
  (0, _react.useEffect)(() => {
    if (liked !== undefined) {
      setLike(liked);
    }
  }, [liked]);
  const content = /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: `${prefixCls}-content`,
      children: [type !== 'pithy' && tokenId !== undefined ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-serial-number`,
        children: `#${tokenId}`
      }) : null, typeof image === 'string' ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Image, {
        width: "100%",
        src: (0, _web3Common.getWeb3AssetUrl)(image),
        ...antdImageProps
      }) : image]
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: `${prefixCls}-body`,
      children: [tokenId !== undefined && type === 'pithy' ? /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: `${prefixCls}-serial-number`,
        children: ["No:", tokenId.toString()]
      }) : null, name ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-name`,
        children: name
      }) : null, description ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-description`,
        children: description
      }) : null, /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: `${prefixCls}-info`,
        children: [price ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: `${prefixCls}-price`,
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_cryptoPrice.CryptoPrice, {
            ...price
          })
        }) : null, likeConfig ? /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
          className: `${prefixCls}-likes`,
          children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
            className: (0, _classnames.default)(`${prefixCls}-like-icon`, {
              [`${prefixCls}-like-icon-liked`]: like
            }),
            onClick: handleLikeChange,
            children: [iconLikeGroup, !like ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_icons.default, {
              component: HeartSvg,
              className: `${prefixCls}-like-icon-icon-heart`
            }) : null]
          }), /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
            className: `${prefixCls}-like-value`,
            children: (0, _tool.formatNumUnit)(totalLikes)
          })]
        }) : null]
      }), showAction ? /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-action`,
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Button, {
          onClick: onActionClick,
          children: actionText ?? messages.actionText
        })
      }) : null, footer ? /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: `${prefixCls}-footer`,
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Divider, {
          dashed: true
        }), footer]
      }) : null]
    })]
  });
  return wrapSSR( /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    className: mergeCls,
    style: style,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: `${prefixCls}-inner`,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(CardSkeleton, {
        prefixCls: `${prefixCls}-skeleton`,
        loading: loading,
        children: content
      })
    })
  }));
};
var _default = exports.default = NFTCard;
