// @ant-design/web3/dist/esm/connect-modal/components/MainPanel.js
import React, { useContext } from 'react';
import { connectModalContext } from "../context";
import DefaultGuidePanel from "./DefaultGuidePanel";
import GetWalletPanel from "./GetWalletPanel";
import QrCode from "./QrCode";
import WalletCard from "./WalletCard";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var MainPanel = function MainPanel(props) {
  var guide = props.guide,
    walletList = props.walletList,
    simple = props.simple;
  var _useContext = useContext(connectModalContext),
    prefixCls = _useContext.prefixCls,
    panelRoute = _useContext.panelRoute,
    selectedWallet = _useContext.selectedWallet;
  return /*#__PURE__*/_jsxs("div", {
    className: "".concat(prefixCls, "-main-panel"),
    children: [panelRoute === 'guide' || panelRoute === 'init' ? /*#__PURE__*/_jsx(DefaultGuidePanel, {
      guide: guide
    }) : null, panelRoute === 'getWallet' ? /*#__PURE__*/_jsx(GetWalletPanel, {
      walletList: walletList
    }) : null, panelRoute === 'wallet' && selectedWallet ? /*#__PURE__*/_jsx(WalletCard, {}) : null, panelRoute === 'qrCode' && selectedWallet ? /*#__PURE__*/_jsx(QrCode, {
      wallet: selectedWallet,
      simple: simple
    }) : null, panelRoute === 'downloadQrCode' && selectedWallet ? /*#__PURE__*/_jsx(QrCode, {
      wallet: selectedWallet,
      simple: simple,
      download: true
    }) : null]
  });
};
export default MainPanel;
