// @ant-design/web3/dist/lib/utils/test-utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.browsers = void 0;
exports.mockBrowser = mockBrowser;
exports.mockClipboard = void 0;
const oriClipboard = window.navigator.clipboard;
// 代理 navigator.clipboard 方法用于测试复制文本到粘贴板的功能
const mockClipboard = () => {
  const clipboard = {
    text: '',
    writeText: text => {
      clipboard.text = text;
      return Promise.resolve();
    },
    readText: () => {
      return Promise.resolve(clipboard.text);
    }
  };
  Object.defineProperty(window, 'navigator', {
    value: {
      clipboard
    }
  });
  return () => {
    Object.defineProperty(window, 'navigator', {
      value: {
        clipboard: oriClipboard
      }
    });
  };
};
exports.mockClipboard = mockClipboard;
const browsers = exports.browsers = {
  Chrome: 'chrome',
  Edge: 'edge',
  Firefox: 'firefox',
  Safari: 'safari'
};
function mockBrowser(browser) {
  const oriUserAgent = window.navigator.userAgent;
  Object.defineProperty(window, 'navigator', {
    value: {
      userAgent: browsers[browser]
    }
  });
  return () => {
    Object.defineProperty(window, 'navigator', {
      value: {
        userAgent: oriUserAgent
      }
    });
  };
}
